pkg/steps/template.go

```diff
@@ -382,21 +382,23 @@ func isPodCompleted(podClient coreclientset.PodInterface, name string) (bool, er
 }
 
 func waitForTemplateInstanceReady(templateClient templateclientset.TemplateInstanceInterface, instance *templateapi.TemplateInstance) (*templateapi.TemplateInstance, error) {
-	for {
-		ready, err := templateInstanceReady(instance)
-		if err != nil {
-			return nil, fmt.Errorf("could not determine if template instance was ready: %v", err)
+	var actualErr error
+	err := wait.PollImmediate(2*time.Second, 10*time.Minute, func() (bool, error) {
+		instance, actualErr = templateClient.Get(instance.Name, meta.GetOptions{})
+		if actualErr != nil {
+			return false, nil
 		}
+		ready := false
+		ready, actualErr = templateInstanceReady(instance)
 		if ready {
-			return instance, nil
-		}
-
-		time.Sleep(2 * time.Second)
-		instance, err = templateClient.Get(instance.Name, meta.GetOptions{})
-		if err != nil {
-			return nil, fmt.Errorf("unable to retrieve existing template instance: %v", err)
+			return true, nil
 		}
+		return false, nil
+	})
+	if err == nil {
+		return instance, nil
 	}
+	return nil, actualErr
```

Comment:
it seemed better to no return an instance ptr where it existed but was not ready ... the calling code does not care, but I was worried about possible future use